index.js

```diff
@@ -70,22 +70,40 @@ function build (schema, options) {
 
   var main
 
+  // wrapper function for processing nullable values
+  var processValue = (schema, code, name, nonNullHandler) => {
+    code += `
+      function ${name} (input) {
+        if(input === null) {
+          return null
+        } else {
+          return ${nonNullHandler}(input)
+        }
+      }
+    `
+    return code
+  }
+
   switch (schema.type) {
     case 'object':
       main = '$main'
       code = buildObject(schema, code, main, options.schema, schema)
       break
     case 'string':
-      main = $asString.name
+      main = schema.nullable ? '$main' : $asString.name
+      code = schema.nullable ? processValue(schema, code, main, $asString.name) : code
       break
     case 'integer':
-      main = $asInteger.name
+      main = schema.nullable ? '$main' : $asInteger.name
+      code = schema.nullable ? processValue(schema, code, main, $asInteger.name) : code
       break
     case 'number':
-      main = $asNumber.name
+      main = schema.nullable ? '$main' : $asNumber.name
+      code = schema.nullable ? processValue(schema, code, main, $asNumber.name) : code
       break
     case 'boolean':
-      main = $asBoolean.name
+      main = schema.nullable ? '$main' : $asBoolean.name
+      code = schema.nullable ? processValue(schema, code, main, $asBoolean.name) : code
       break
```

Comment:
Given that these are only for standard types, should we avoid generating a single function for each property but rather use some functions that we add? Something like: $asBooleanNullable, $asNumberNullable, $asStringNullable, etc..